### PR TITLE
Adds basic {de,}serialization verbs for admins

### PR DESCRIPTION
**Pull request by Crazylemon64**
Admins can toy around with converting a character to text, and
placing them back elsewhere, independent of the round.

It's got a kink, though: The text given to you directly needs to be
printed once to remove all the escapes - otherwise the thing will choke.
I'd like advice on resolving that.

By "printing", I mean pasting the text into a python console or similar, and feeding it to a "print" operation - the resulting text will work fine.

Also, currently the only objects with a real serialization response are humans and their organs - doing EVERYTHING would be a monolithic task - most things will just instance a new object from their path.

The average human is about 2KB of JSON text - Each organ with a unique DNA datum will have their own entry in the data, however, and many can really bump up the size.
:cl:Crazylemon
rscadd: Admins now have serialization verbs for storing things as text between rounds
rscadd: Sophisticated serialization support for humans, organs, limbs (not equipment)
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -2031,3 +2031,59 @@
...
+			continue
+		// "New" code handles insertion and DNA sync'ing
+		var/obj/item/organ/external/E = list_to_object(limbs_list[limb], src)
+		E.sync_colour_to_dna()
+
+	for(var/organ in organs_list)
+		// As above, "New" code handles insertion, DNA sync
+		list_to_object(organs_list[organ], src)
+	update_icons()
+	..()
```
Comment (Crazylemon64) [inline code/modules/mob/living/carbon/human/human.dm]: :-1: :eyes:

**Comment by monster860** [comment]
@Crazylemon64 `html_decode()` ?

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -2031,3 +2031,59 @@
...
+			limbs_list[limb] = "missing"
+			continue
+
+		limbs_list[limb] = O.serialize()
+
+	for(var/organ in internal_organs)
+		var/obj/item/organ/O = organ
+		organs_list[O.name] = O.serialize()
+
+	return json_encode(data)
```
Comment (monster860) [inline code/modules/mob/living/carbon/human/human.dm]: `return data`

File: code/modules/admin/verbs/serialization.dm [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+	if(!check_rights(R_ADMIN))
+		return
+
+	if(!istype(holder.marked_datum, /atom/movable))
+		to_chat(src, "The marked datum is not an atom/movable!")
+		return
+
+	var/atom/movable/AM = holder.marked_datum
+	to_chat(src, json_encode(AM.serialize()))
```
Comment (monster860) [inline code/modules/admin/verbs/serialization.dm]: And keep this the same!

**Comment by monster860** [comment]
You're running `json_encode` twice, which is why you're ending up with an escaped string.

**Comment by Crazylemon64** [comment]
Hold on, a github hiccup swallowed a commit

**Pull request closed without merging**